hack/make/go.mk

```diff
@@ -52,6 +52,6 @@ go/gen_mocks: prerequisites/mockery
 	mockery
 
 ## Runs deadcode https://go.dev/blog/deadcode
-go/deadcode:
```

Comment:
Any reason why this is not called within `go/lint`?

Reply:
we are not using go/lint in CI, we use github action ... it was like this before I think.

Reply:
That's ok, would still be nice if you call `go/deadcode` within the `go/lint` command so it automatically runs as soon as we lint

Reply:
do you mean this b0d4b8bf?

Reply:
Ups, didn't see that 👍

Reply:
Well, that commit was on the wrong branch 😉

Reply:
fixed. :)